src/plugins/intel_gpu/src/kernel_selector/kernels/softmax/softmax_kernel_bf.cpp

```diff
@@ -54,9 +54,9 @@ SoftmaxKernel_bf::Parent::DispatchData SoftmaxKernel_bf::SetDefault(const softma
         }
 
         dispatchData.leftovers = dispatchData.dataSetSize % dispatchData.lws[0];
-        //if (dispatchData.leftovers % subgroup_size) {
         // To use subgroup read/write, the starting address should be aligned to 128 bit
-        if ((dispatchData.dataSetSize * params.inputs[0].ElementSize()) >> 4) {
+        size_t dataSetSizeInByte = dispatchData.dataSetSize * params.inputs[0].ElementSize();
+        if (dispatchData.dataSetsCount == 1 || (dataSetSizeInByte - ((dataSetSizeInByte >> 4) << 4))) {
```

Comment:
Why blocking read is disabled for a single dataset input?

I'd propose the following changes for better for code readability.
```C
if (dispatchData.dataSetsCount != 1 && dataSetSizeInByte % 16) {
```

Reply:
dataSetCount is my fault :( My intention was the opposite. Thanks!
BTW I want to reduce the computation as much as possible because SEtDefault is called at runtime update at dynamic shape.